libadikted/Makefile

```diff
@@ -1,14 +1,50 @@
 # Project: libadikted
+# Makefile created by KeeperFX/dkfans team
 
-#To create a debug executable, run with the command:
-# make DEBUG
+# Instructions:
+# To create the Windows library run:
+# make all
+# To create the linux library run:
+# make all BUILD_FOR_LINUX=1 (not fully implemented)
+# To create a debug library, run with the command:
+# make DEBUG (THIS IS NOT IMPLEMENTED YET)
+# To clean up the built files (objects, library, and resource file) run:
+# make clean
 
-#CROSS_COMPILE = i686-w64-mingw32-
+# Tested and working with:
+# - Windows : MSYS2 (MinGW 32-bit) - make
+# - Windows : MSYS2 (MinGW 32-bit) - mingw32-make
+# - Windows : MinGW (MSYS) - make
 
-
-CC           = $(CROSS_COMPILE)gcc
-LINK         = $(CROSS_COMPILE)gcc
-SRC          = \
+# Executable files extension on host environment
+ifneq (,$(findstring Windows,$(OS)))
+  CROSS_EXEEXT = .exe
+else
+  CROSS_EXEEXT =
+  CROSS_COMPILE = i686-w64-mingw32-
+endif
+# Executable files extension on target environment (
+#BUILD_FOR_LINUX = 1
+ifneq (,$(BUILD_FOR_LINUX))
+  LIBEXT = .so
+  OTHERLIBEXT = .dll
+else
+  LIBEXT = .dll
+  OTHERLIBEXT = .so
+endif
+# Names of utility commands
+CPP                = $(CROSS_COMPILE)g++$(CROSS_EXEEXT) -std=c89
+CC                 = $(CROSS_COMPILE)gcc$(CROSS_EXEEXT) -std=c89 -Wall -Wextra -Wno-pedantic -Wno-conversion -Wno-traditional-conversion -Wno-sign-compare
+CFLAGS             = -c -O -I. -DBUILD_DLL -fexpensive-optimizations -O3 -march=pentium-mmx -mmmx
```

Comment:
`-fexpensive-optimizations -O3` 
`-march=pentium-mmx -mmmx`

Through my ignorance, I want to double check these flags are wanted/needed? The mmx ones are used just below too.

Reply:
You want to keep the optimization flags `-fexpensive-optimizations` and `-O3`, should only change those if benchmarks shows different optimizations to be faster.

As for `-march=pentium-mmx` and `-mmmx`, these have always been part of the compilation flags since 0.95 release, `-march` was set to `i486` prior to that. If you remove `-march` flag then GCC will use a default value which currently is `i686` which lacks MMX instruction support, so I'd just keep it unless we want to have it run on some even older CPU's.

Reply:
It might be worth to not have it use the optimization flags `-fexpensive-optimizations` and `-O3` in case you do a debug build.

Reply:
I think you should check if `-march` and `-mmmx` are still required for gcc 9+

Reply:
@SimLV they are not at all required. As I said in my previous comment the default value for `-march` is `i686` for 32-bit builds. Removing both `-march` and `-mmmx` would prevent the compiler from using the MMX CPU instructions, not sure why you'd want that unless you want to run it on pre-1997 CPU's?